### PR TITLE
Action group compact

### PR DESCRIPTION
**Pull request by snowystinger**
Closes https://jira.corp.adobe.com/browse/RSP-1808

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [Issue](https://github.com/adobe-private/react-spectrum-v3/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

**Comment by solimant** [comment]
@snowystinger - wanna merge latest from `master` to bring this branch up to speed?

File: packages/@react-spectrum/actiongroup/stories/ActionGroup.stories.tsx [hunk 1]
```diff
@@ -156,3 +171,184 @@ function render(props = {}, items: any = itemsWithIcons) {
...
+              <IconElement />
+            </Item>
+          );
+        })
+      }
+    </ActionGroup>
+  );
+}
+
+function renderBigOne() {
```
Comment (solimant) [inline packages/@react-spectrum/actiongroup/stories/ActionGroup.stories.tsx:222]: 🤯 
Although this is just a story, I'm think the `renderThing()` way of doing things is a React anti-pattern, and should probably be taken out into its own component.

Comment (snowystinger) [reply packages/@react-spectrum/actiongroup/stories/ActionGroup.stories.tsx:222]: yeah, for whatever reason it's the storybook convention we've always had even back in v2
so unless you have an immediate concern, I'm probably not refactoring this in this pr

Comment (solimant) [reply packages/@react-spectrum/actiongroup/stories/ActionGroup.stories.tsx:222]: yes, I realized that after.. definitely no concerns.. in fact, probably no biggie for stories in general

**Comment by ktabors** [comment]
When testing I click between the buttons and I'm able to get a focus ring around the entire group.
Chrome
image
Firefox
image

**Comment by ktabors** [comment]
Outside of scope, but should docs cover `density` and `disabledKeys`? I don't think `isJustified` needs to be covered.

File: packages/@react-spectrum/actiongroup/stories/ActionGroup.stories.tsx [hunk 2]
```diff
@@ -156,3 +171,184 @@ function render(props = {}, items: any = itemsWithIcons) {
...
+  return (
```
[the remark this answers is not in the document]
Comment (snowystinger) [reply packages/@react-spectrum/actiongroup/stories/ActionGroup.stories.tsx:223]: this only matches the XD file
https://spectrum-contributions.corp.adobe.com/page/action-group-beta/

**Pull request merged**